src/v/kafka/server/requests.cc

```diff
@@ -126,15 +126,16 @@ handle_auth_handshake(request_context&& ctx, ss::smp_service_group g) {
          * is set the input connection is assumed to contain raw authentication
          * tokens not wrapped in a normal kafka request envelope.
          */
-        conn->sasl().set_handshake_v0();
+        conn->sasl()->set_handshake_v0();
     }
     return do_process<sasl_handshake_handler>(std::move(ctx), g)
       .response.then([conn = std::move(conn)](response_ptr r) {
-          if (conn->sasl().has_mechanism()) {
-              conn->sasl().set_state(
+          if (conn->sasl()->has_mechanism()) {
+              conn->sasl()->set_state(
```

Comment:
i guess conn->sasl() is checked to have value further up the call chain (since we aren't using .value() or asserts in here)?

Reply:
Yep: https://github.com/BenPope/redpanda/blob/a8f4e4f9a70fbcd6f98c4b7c4622778450e61faf/src/v/kafka/server/requests.cc#L253